src/test/java/org/rulez/demokracia/pdengine/BeatTableBeatInformationTest.java

```diff
@@ -0,0 +1,101 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.BeatTable.Direction;
+import org.rulez.demokracia.pdengine.annotations.TestedBehaviour;
+import org.rulez.demokracia.pdengine.annotations.TestedFeature;
+import org.rulez.demokracia.pdengine.annotations.TestedOperation;
+import org.rulez.demokracia.pdengine.dataobjects.Pair;
+import org.rulez.demokracia.pdengine.exception.IsNullException;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultChoice;
+
+public class BeatTableBeatInformationTest extends CreatedDefaultChoice {
+
+	@Before
+	public void setUp() throws ReportedException {
+		super.setUp();
+	}
+
+	@TestedFeature("Supporting functionality")
+	@TestedOperation("BeatTable")
+	@TestedBehaviour("the beat information related to a and b can be obtained for forward and backward")
+	@Test
+	public void beatInformation_needs_a_defined_direction() {
+		BeatTable beatTable = new BeatTable();
+
+		assertThrows(() -> beatTable.beatInformation(null, null, null)
+				).assertMessageIs("Invalid direction");
+	}
+	
+	@TestedFeature("Supporting functionality")
+	@TestedOperation("BeatTable")
+	@TestedBehaviour("the beat information related to a and b can be obtained for forward and backward")
+	@Test
+	public void The_1st_key_should_be_a_Choice() {
+		BeatTable beatTable = new BeatTable();
+		
+		assertThrows(() -> beatTable.beatInformation(null, null, Direction.DIRECTION_FORWARD)
+				).assertMessageIs("Invalid row key");
+	}
+	
+	@TestedFeature("Supporting functionality")
+	@TestedOperation("BeatTable")
+	@TestedBehaviour("the beat information related to a and b can be obtained for forward and backward")
+	@Test
+	public void beatTable_with_empty_matrix() {
+		Choice choice1 = new Choice("name", "userName");
+		Choice choice2 = null;
+		
+		BeatTable beatTable = new BeatTable();
+		
+		assertThrows(() -> beatTable.beatInformation(choice1, choice2, Direction.DIRECTION_FORWARD)
+				).assertMessageIs("Invalid row key");
+	}
+	
+	@TestedFeature("Supporting functionality")
+	@TestedOperation("BeatTable")
+	@TestedBehaviour("the beat information related to a and b can be obtained for forward and backward")
+	@Test
+	public void beatTable_forward() {
```

Comment:
google translate gives and acceptable translation for "beatInformation forward irányban megadja choice1 choice2-vel szembeni nyeréseinek számát "